File: OrchidCore/src/main/java/com/eden/orchid/api/options/globals/DataGlobal.java
package com.eden.orchid.api.options.globals;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.TemplateGlobal;

import java.util.Map;

public class DataGlobal implements TemplateGlobal<Map<String, Object>> {

    @Override
    public String key() {
        return "data";
    }

    @Override
    public Map<String, Object> get(OrchidContext context) {
        return context.getData().toMap();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/globals/DataGlobal.kt
package com.eden.orchid.api.options.globals

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.TemplateGlobal
import com.eden.orchid.api.theme.pages.OrchidPage

class DataGlobal : TemplateGlobal<Map<String?, Any?>?> {
    override fun key() = "data"
    override fun get(context: OrchidContext, page: OrchidPage?): Map<String?, Any?>? = context.data
}
